glotaran/model/model.py

```diff
@@ -312,13 +312,39 @@ def as_dict(self) -> dict:
 
         return model_dict
 
-    def get_parameters(self) -> list[str]:
-        parameters = []
+    def get_parameter_labels(self) -> list[str]:
+        parameter_labels = []
         for item_name in self.model_items:
             items = getattr(self, item_name)
             item_iterator = items if isinstance(items, list) else items.values()
             for item in item_iterator:
-                parameters += item.get_parameters()
+                parameter_labels += item.get_parameter_labels()
+        return parameter_labels
+
+    def generate_parameters(self) -> dict | list:
+        parameters: dict | list = {}
```

Comment:
A unit test has been added.

The typing I cannot fix now since I don't now how to create types referencing themselves. I created #1020 to work on this in a future PR.